src/webview/html/messageHeaderAsHtml.js

```diff
@@ -57,45 +60,49 @@ export default (
     const topicHtml = renderSubject(item);
 
     return template`
-<div class="header-wrapper stream-header" data-msg-id="${item.id}">
+<div class="header-wrapper header stream-header" data-msg-id="${item.id}">
   <div class="header stream-text"
        style="color: ${textColor};
               background: ${backgroundColor}"
        data-narrow="${streamNarrowStr}">
     # ${item.display_recipient}
   </div>
-  <div class="header topic-text" data-narrow="${topicNarrowStr}">
-    $!${topicHtml}
+  <div class="header-wrapper header topic-content" data-narrow="${topicNarrowStr}">
+    <div class="topic-text">$!${topicHtml}</div>
+    <div class="human-date">$!${humanDate(new Date(item.timestamp * 1000))}</div>
   </div>
 </div>
     `;
   }
 
-  if (
-    item.type === 'private'
-    && !isPrivateNarrow(narrow)
-    && !isGroupNarrow(narrow)
-    && !isTopicNarrow(narrow)
-  ) {
-    const recipients =
-      item.display_recipient.length === 1 && item.display_recipient[0].email === ownEmail
-        ? item.display_recipient
-        : item.display_recipient.filter(r => r.email !== ownEmail);
+  if (item.type === 'private') {
+    let recipientsStr;
+    let privateNarrowStr;
```

Comment:
I don't like these mutable variables -- they're a smell, a sign that something's off in the logic.

One observation: you're including the div for "recipients" even when there's nothing there. That seems likely to cause trouble -- if nothing else, confusion.

Probably better to just have a separate case for the new situation where we're showing this header but don't want to show the recipients.

Reply:
I agree with this. Introduced another case instead of checking with variables (now checking the narrow with `isPrivateNarrow() && !isSpecialNarrow()` to check for if the screen the user is on is a private conversation.)